plugins/modules/foreman_search_facts.py

```diff
@@ -107,16 +123,18 @@ def main():
             resource=dict(type='str', required=True),
             search=dict(default=""),
             full_details=dict(type='bool', aliases=['info'], default='false'),
+            params=dict(type='dict', default=dict()),
         ),
     )
 
     module_params = module.clean_params()
     resource = module_params['resource']
     search = module_params['search']
+    params = module_params['params']
```

Comment:
```suggestion
    params = module_params.get('params')
```

that way it will really be None if the user didn't pass it.